social/backends/trinity.py

```diff
@@ -0,0 +1,54 @@
+import json
+
+from social.backends.oauth import BaseOAuth2
+from urllib import urlencode
+
+class TrinityOauth2(BaseOAuth2):
+    name = 'trinity'
+    REDIRECT_STATE = False # ????
+    AUTHORIZATION_URL = 'https://pass.texasgateway.org/oauth/v2/auth'
+    ACCESS_TOKEN_URL = 'https://pass.texasgateway.org/oauth/v2/token'
+    ACCESS_TOKEN_METHOD = 'POST'
+    SCOPE_SEPARATOR = ','
+    #DEFAULT_SCOPE = ['email']
+    EXTRA_DATA = [
+        ('email', 'email'),
+        ('username', 'username'),
+        ('fullname', 'fullname'),
+        ('district', 'district'),
+    ]
+
+    def get_user_details(self, response):
+        """Return user details from GitHub account"""
+        fullname, first_name, last_name = self.get_user_names(
+            '', response.get('first_name', ''),response.get('last_name', '')
+        )
+        data = {'username': response.get('username'),
```

Comment:
Add a fallback value here, or, if `username` is required and not provided, we should throw an error.